code/modules/projectiles/projectile/special/rocket.dm

```diff
@@ -39,67 +38,77 @@
+	damage = 80
+	armour_penetration = 100
+	dismemberment = 100
+	anti_armour_damage = 200
+
+/obj/projectile/bullet/rocket/heap/do_boom(atom/target, blocked=0)
+	explosion(target, devastation_range = -1, heavy_impact_range = 1, light_impact_range = 3, flame_range = 4, flash_range = 1, adminlog = FALSE)
+
+/// PM9 weak rocket - just kind of a failure
+/obj/projectile/bullet/rocket/weak
+	name = "low-yield rocket"
 	desc = "Boom, but less so."
 	damage = 30
-	anti_armour_damage = 0
 
-/obj/projectile/bullet/a84mm/weak/do_boom(atom/target, blocked=0)
+/obj/projectile/bullet/rocket/weak/do_boom(atom/target, blocked=0)
 	if(!isliving(target)) //if the target isn't alive, so is a wall or something
 		explosion(target, heavy_impact_range = 1, light_impact_range = 2, flame_range = 3, flash_range = 4, explosion_cause = src)
 	else
 		explosion(target, light_impact_range = 2, flame_range = 3, flash_range = 4, explosion_cause = src)
 
-/// Mech BRM-6 missile
-/obj/projectile/bullet/a84mm_br
-	name ="\improper HE missile"
-	desc = "Boom."
-	icon_state = "missile"
+/** SRM-8 Missile - Used by the SRM-8 Exosuit missile rack.
+* Employed by Nuclear Operatives Maulers and Nanotrasen Marauders and Seraphs to kill everything and anyone.
+*
+* Explodes when it hits literally anything.
+*/
+/obj/projectile/bullet/rocket/srm
+	name = "short range missile"
+	desc = "Today's not your day, pal."
+
+/** PEP-6 Missile - Used by the PEP-6 Exosuit missile rack.
+* Employed by Roboticists out of spite to put down enemy hereteks, mechanized nuclear operatives, the janitor's hot rod,
+* the clown's 'taxi service', uppity borgs, vengeful ais, doors they don't like, the escape shuttle's hull, and more!
+*
+* Explodes only when it hits specifically one of the following types:
+* (/obj/structure), (/obj/machinery), (/obj/vehicle), (/turf/closed), (/mob/living/silicon)
+*
+* Does NOT explode if it hits any random mob, or any random object. Only if it is a subtype of one of the above valid atoms.
+*/
+/obj/projectile/bullet/rocket/pep
+	name = "precise explosive missile"
+	desc = "Human friendly, metal unfriendly."
 	damage = 30
-	ricochets_max = 0 //it's a MISSILE
-	embedding = null
-	shrapnel_type = null
-	var/sturdy = list(
-	/turf/closed,
-	/obj/vehicle/sealed/mecha,
-	/obj/machinery/door,
-	/obj/structure/window,
-	/obj/structure/grille
-	)
+	anti_armour_damage = 80 //Doesn't (probably) kill borgs in one shot, but it will hurt
+	random_crits_enabled = FALSE //yeah, no
+
+/obj/projectile/bullet/rocket/pep/do_boom(atom/target, blocked=0)
+	if(issilicon(target)) //if the target is a borg, just give them one of these to make it loud, most of the damage is in the projectile itself
+		explosion(target, light_impact_range = 1, flash_range = 2, explosion_cause = src)
+		return
+	if(isstructure(target) || isvehicle (target) || isclosedturf (target) || ismachinery (target)) //if the target is a structure, machine, vehicle or closed turf like a wall, explode that shit
```

Comment:
make a weaker explosion or something if its not dense, light fixtures or something probably shouldnt cause a heavy impact